SemanticSearch2.md

```diff
@@ -0,0 +1,320 @@
+					"name": "extracted keyword job level",
+					"weight": 1.0
+				}],
+				"text_kernel_related_keywords": [{
+					"name": "extracted keyword Text Kernel related keyword",
+					"weight": 1.0
+				}]
+			}
+		}],
+		"versions": {
+			"interesting_keywords": "InterestingTermsV2",
+			"job_level": "JobLevel",
+			"job_titles": "CaroteneV3",
+			"occupations": "ONet17",
+			"raw_job_titles_jobseeker": "RawJobTitlesJobSeekerV1",
+			"raw_job_titles_recruiter": "RawJobTitlesRecruiterV1",
+			"related_keywords": "RelatedSearchTermsV1",
+			"related_keywords_jobseeker": "RelatedSearchTermsJobseekerV1",
+			"related_keywords_recruiter": "RelatedSearchTermsRecruiterV1",
+			"skills": "SkillsV4",
+			"text_kernel_related_keywords": "TextKernelRelatedSearchTerms",
+			"custom_keywords": "CustomKeywords"
+		}
+	}
+}
+```
+
+# Document Description
+
+Used to parse a job or resume into its most important features and to represent these as a prioritized list of features that can be used to form a matching query.
+
+# Document Request Information
+
+HTTP method: GET or POST
+Parameters (query/form):
+-	language (optional) : two letter language code followed by underscore, followed by two letter country code. Determines the language in which enrichments are returned. Required to use personalized overrides. Defaults to en_us. Currently, allowed values are en_us, en_gb, fr_fr, de_de, and nl_nl.
+-	user_id (optional) : user id for which to look up personalized overrides. Defaults to null. 
+-	relationships_threshold : the mimimum weight for a relationship entry to be added as enrichment. Can be used to prune the result size. Defaults to 0.5. Allowed values are any number between 0 and 1.
+-	document : the binary document to parse as base64 encoded string (according to RFC 3548/4648).
+-	type : the type of document either "JOB" or "RESUME".
+
+Example 2.0 request:
+```
+HTTP GET
+Accept: application/json;version=2.0
+https://api.careerbuilder.com/core/semanticsearch/document
+{
+	"type": "JOB",
+	"document": "SmF2YSBkZXZlbG9wZXIKCkNvbXBhbnk6IENhcmVlckJ1aWxkZXIKTG9jYXRpb246IEF0bGFudGEsIEdlb3JnaWEKCgpBcHBseSBub3chIFRoYW5rIHlvdSBmb3IgYXBwbHlpbmcuCg==",
+	"relationships_threshold": "0.9"
+}
+```
+
+# Document Response
+
+The document response is divided into two parts. First is the parsed_input node which contains information on all extracted keywords and phrases. Next is the extracted_keywords node, which gives the name, type and relationships of all extracted phrases:
+
+- data
+	- parsed_input
+		- input (comma-separated list of extracted keywords and phrases)
+		- parsed_fragments (input as list of strings)
+		- input_to_extracted_keywords (map of extracted strings to canonical strings)
+	- extracted_keywords (list)
+		- name (canonical string)
+		- weight (float)
+		- type (string: skill, job_title, keyword, company, school, location, company_geography)
+		- entity_node (object of type-specific attributes for type: location and company_geography)
```

Comment:
That is a good point, and the same goes for the list of relationships per type.

Is it possible to define all entity_node attributes and relationships per type now? Will this no longer change?